ui/codeView.js

```diff
@@ -1411,6 +1408,16 @@ var CodeViewManager = GObject.registerClass({
         });
     }
 
+    removeSessions() {
+        this._sessions.forEach(session => {
+            // flip back the app actor
+            session.app.rotation_angle_y = 0;
+            session._setEffectsEnabled(session.app, false);
+            const eventType = SessionDestroyEvent.SESSION_DESTROY_APP_DESTROYED;
+            this._removeSession(session, eventType);
```

Comment:
I am not completely sure whether you can update the array (`this._sessions`) while iterating. Maybe more safe to do something like:
```
for (i = this._sessions.length - 1; i >= 0; --i) {
   let session = this._sessions[i];
   ...
   this._removeSession(...);
}
```

Reply:
Pushed a fixup commit for that, please check if all good. I suggest we squash this commit before merging this branch.

Reply:
It is safe to splice the array inside a forEach call, both ways are fine.

Reply:
I see, but I've seen some comments in the interwebs (e.g. https://stackoverflow.com/questions/9882284/looping-through-array-and-removing-items-without-breaking-for-loop) that splicing will re-index the array and skip over an index when one is removed... In any case I think it is safer to use length for that instead.

Reply:
If the array is modified during the `forEach` the loop doesn't iterate over the whole array, the `forEach` loop maintains the index, but if the array changes, that index couldn't be the same position, so I think this solution is better.